### PR TITLE
add rocm_3.7 to the nightly binary build

### PR DESCRIPTION
**Pull request by walterddr**

### CODE REVIEW

File: .circleci/cimodel/data/binary_build_definitions.py [hunk 1]
```diff
@@ -37,9 +37,10 @@ def gen_docker_image(self):
         docker_distro_prefix = miniutils.override(self.pydistro, docker_word_substitution)
 
         # The cpu nightlies are built on the pytorch/manylinux-cuda102 docker image
-        alt_docker_suffix = self.cuda_version or "102"
-        docker_distro_suffix = "" if self.pydistro == "conda" else alt_docker_suffix
-        return miniutils.quote("pytorch/" + docker_distro_prefix + "-cuda" + docker_distro_suffix)
+        alt_docker_suffix = self.gpu_version or "cuda102"
+        docker_distro_suffix = alt_docker_suffix if self.pydistro != "conda" else (
+            "cuda" if alt_docker_suffix.startswith("cuda") else "rocm")
+        return miniutils.quote("pytorch/" + docker_distro_prefix + "-" + docker_distro_suffix)
```
Comment (jeffdaily) [inline .circleci/cimodel/data/binary_build_definitions.py:43]: @seemethere requested we have one ROCm docker repo `pytorch/manylinux-rocm` and use tags for versions.  Example, `pytorch/manylinux-rocm:3.7`.

**Comment by jeffdaily** [comment]
Since we do not have ROCm-based circleci resources, I think we also need to somehow exclude ROCm from `binary_linux_test` since there is no means of testing.  Instead, we just need build and upload.  Right?  CC @seemethere

**Comment by jeffdaily** [comment]
We need to update the file `.circleci/scripts/binary_linux_build.sh` and update it perhaps like this:
```diff
diff --git a/.circleci/scripts/binary_linux_build.sh b/.circleci/scripts/binary_linux_build.sh
index 5cb7515..5355a1f 100755
--- a/.circleci/scripts/binary_linux_build.sh
+++ b/.circleci/scripts/binary_linux_build.sh
@@ -12,6 +12,8 @@ if [[ "$PACKAGE_TYPE" == 'conda' ]]; then
   build_script='conda/build_pytorch.sh'
 elif [[ "$DESIRED_CUDA" == cpu ]]; then
   build_script='manywheel/build_cpu.sh'
+elif [[ "$DESIRED_CUDA" == *rocm* ]]; then
+  build_script='manywheel/build_rocm.sh'
 else
   build_script='manywheel/build.sh'
 fi
```
There will be a corresponding change in https://github.com/pytorch/builder/pull/503 to add this `build_rocm.sh` script.  It would differ too greatly from the `build.sh` to have a single file, which is why I'm adding it.

File: .circleci/cimodel/data/binary_build_definitions.py (hunk 1 above)
Comment (walterddr) [reply .circleci/cimodel/data/binary_build_definitions.py:43]: @seemethere this is actually something we should also probably do for cuda too? at least for the minor versions?

**Comment by walterddr** [comment]
closing in favor of #44250 since this fork doesn't work with github tags

**Pull request closed without merging**